### PR TITLE
Feature 431. Migrate to AG Grid and direct streaming

### PR DESCRIPTION
**Pull request by MytsV**
- Developed a hook for streaming data directly and covered it with unit test
- Developed a hook for throttling data (may be useful when implementing the dashboard)
- Styled the AG Grid to fit the general theme of the app
- Replaced the ComDOM implementation with the direct streaming for the RSE list page as a demo

There are still a lot of pending improvements to the table and generally the UI. I need a little time to come up with a new design to make more changes to the responsiveness and appearance.

### CODE REVIEW

File: src/component-library/tailwind.css [hunk 1]
```diff
@@ -7,4 +7,70 @@
         font-size: 16px;
         /* font-family: monospace; */
     }
-}
\ No newline at end of file
+}
+
+/*@layer base {*/
```
Comment (maany) [inline src/component-library/tailwind.css:12]: @MytsV any reason to keep these commented css styles here?

Comment (maany) [reply src/component-library/tailwind.css:12]: Are storybooks able to pick up this css file?

Comment (MytsV) [reply src/component-library/tailwind.css:12]: Sorry, that's still a WIP and something got left from connecting shadcn/ui. I'll review it when working on the toasts in this PR.

> @MytsV any reason to keep these commented css styles here?

Comment (MytsV) [reply src/component-library/tailwind.css:12]: > Are storybooks able to pick up the ag-grid-rucio css file?

Yes, and, luckily, without any additional setup!
image

**Comment by MytsV** [comment]
The basic parts for migrating to it everywhere are included here. On top of these changes, pages will be rebuilt 
 and covered by tests independently, as it proved hard to work with the current implementations.

**Comment by MytsV** [comment]
The new version of the streaming is the best of both worlds. While the previous implementation still experienced performance issues with receiving large amounts of data in a short time, ComDOM fetched it fast at least inside the worker. Therefore I decided to try it out and it turned out great. My idea was to make it as basic as possible, so there's less room for bugs.

There are no apparent issues neither with state management nor with loading at any rate. Benchmarks will be set up during the individual work on each page.

**Comment by MytsV** [comment]
As the AG Grid styling doesn't use tailwind, it has to rely on a color mode observer to change it's theme together with tailwind components.

**Comment by MytsV** [comment]
Current light vs dark theme of the table:
image
image

**Comment by MytsV** [comment]
These are all the changes that relate only to the AG Grid integration. To proceed, I need to rewrite the current pages and cover them with tests as they aren't reliable. They will be built on top of this PR + #468 + #466. When the prettier configuration PR is accepted, I'll refactor everything.

**Comment by MytsV** [comment]
I don't like how the code of streamWorker.js turned out and will refactor it, but currently I'm too scared to break something. So I'll cover some page with integration tests first :)

**Comment by MytsV** [comment]
These changes are included in #477

**Pull request closed without merging**